mypy/test/data.py

```diff
@@ -644,40 +648,33 @@ def split_test_cases(
     """
     with open(file, encoding="utf-8") as f:
         data = f.read()
-    # number of groups in the below regex
-    NUM_GROUPS = 7
-    cases = re.split(
-        r"^\[case ([a-zA-Z_0-9]+)"
-        r"(-writescache)?"
-        r"(-only_when_cache|-only_when_nocache)?"
-        r"(-posix|-windows)?"
-        r"(-skip)?"
-        r"(-xfail)?"
-        r"\][ \t]*$\n",
```

Comment:
Previously, when `[case foo-bar-baz]` didn't comply with this pattern, we'd simply skip it and it would end up merging into the section of the preceding `[case]`.